modules/aws/src/smithy4s/query/FormData.scala

```diff
@@ -0,0 +1,55 @@
+package smithy4s.aws.query
+
+final case class FormData(
```

Comment:
That's not bad at all, but I'd rather we represented it as follows (which allows to re-use some types) 

```scala
import smithy4s.PayloadPath

sealed trait FormData {
   def render: String = ??? 
   def prepend(key: String) = ???
   def prepend(index: Int) = ??? 
} 
object FormData {
    final case class SimpleValue(str: String)  extends FormData
    final case class PathedValue(path: PayloadPath, value: String) extends FormData 
    final case class MultipleValues(values: Vector[PathedValue]) extends FormData 
} 
```

The there is no look-up in the encoding side of things, so no need to use a `Map`. You may need to add `def prepend` methods to the existing `PayloadPath` type.

Reply:
Done